src/features/globalSearch/globalSearchSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { searchGloballyFor } from '~/api/globalSearch';
+
+const slice = createSlice({
+  name: 'globalSearch',
+  initialState: {
+    term: '',
+    loading: false,
+    resultItems: null,
+    error: null,
+  },
+  reducers: {
+    setSearchTerm(state, action) {
+      return {
+        ...state,
+        term: action.payload,
+      };
+    },
+    setSearchLoading(state, action) {
+      return {
+        ...state,
+        loading: action.payload,
+      };
+    },
+    setSearchResultItems(state, action) {
+      return {
+        ...state,
+        resultItems: action.payload,
+      };
+    },
+    setSearchFailed(state, action) {
+      return {
+        ...state,
+        error: action.payload,
+      };
+    },
+  },
+});
+
+
+export const searchFor = term => (dispatch) => {
```

Comment:
Sofern ich es korrekt aus der "redux"-Dokumentation entnommen habe, kann und sollte man bei "redux" in `Features` denken.
Die globale Suche wäre somit ein `Feature`, welches eine eigene abgegrenzte (namespace) State-Struktur, einen initialen State, eigene Reducers und davon abgeleitete Actions besitzt. Um die Generierung der Actions kümmert sich das Toolkit selbst, wodurch man diese nicht mehr selbst schreiben muss.

An dieser Stelle definieren wir eine `Thunk`-Funktion, die wir Anstelle einer Action dispatchen können.
Die `Thunk`-Funktion erlaubt uns diverse Actions nacheinander zu dispatchen und auch auf den aktuellen State zuzugreifen.
In diesem Fall wollen wir folgende Aktionen ausführen:

1. Den neuen Suchstring im Store ablegen `dispatch(slice.actions.setSearchTerm(term))`
2. Die Suchergebnisse im Store zurücksetzen, sofern der Suchstring leer ist `dispatch(slice.actions.setSearchResultItems(null))`
3. Den Loading-Status der Suche auf aktiv setzen `dispatch(slice.actions.setSearchLoading(true))`
4. Bei erfolgreichem Such-Request die Suchergebnisse im Store ablegen `dispatch(slice.actions.setSearchResultItems(resultItems))`
5. Im Fehlerfall den Fehlerstring im Store ablegen `dispatch(slice.actions.setSearchFailed(err.toString()))`
6. Abschließend immer den Loading-Status der Suche deaktivieren `dispatch(slice.actions.setSearchLoading(false))`

Reply:
Wird geschlossen, da Redux durch MobX ausgetauscht wurde.